Docker prestissimo worker setup for prestorials

minhancao opened the pull request:
Docker prestissimo worker setup for prestorials:

This Docker setup has:
1. A Presto coordinator to coordinate Presto or Prestissimo workers
2. A Prestissimo worker to run the queries
1. A MinIO container to provide out of the box local S3 storage. 
2. A Hive container to connect MinIO with Presto

wanglinsong commented on running locally/docker-prestissimo/docker-compose.yml:
```diff
@@ -0,0 +1,119 @@
+version: '1'
```
version too old

minhancao replied:
Updated to 3.8 as you have in https://github.com/prestodb/prestorials/blob/main/docker-compose/docker-compose.yaml

wanglinsong commented on running locally/docker-prestissimo/docker-compose.yml:
```diff
@@ -0,0 +1,119 @@
+      AWS_ACCESS_KEY_ID: minioadmin
+      AWS_SECRET_ACCESS_KEY: minioadmin
+    volumes:
+      - '~/minio/data:/minio'
```
home folder?

minhancao replied:
This is where the local data is actually stored for the MinIO container.

wanglinsong commented on running locally/docker-prestissimo/docker-compose.yml:
```diff
@@ -0,0 +1,119 @@
+      AWS_ACCESS_KEY_ID: minioadmin
+      AWS_SECRET_ACCESS_KEY: minioadmin
+    volumes:
+      - './hadoop/core-site.xml:/opt/hadoop/etc/hadoop/core-site.xml'
```
quote or no quote?

wanglinsong commented on running locally/docker-prestissimo/docker-compose.yml:
```diff
@@ -0,0 +1,119 @@
+      - ./prestissimo/catalog/jmx.properties:/opt/presto-server/etc/catalog/jmx.properties
+      - ./prestissimo/catalog/tpcds.properties:/opt/presto-server/etc/catalog/tpcds.properties
+      - ./prestissimo/catalog/tpch.properties:/opt/presto-server/etc/catalog/tpch.properties
+      - ./prestissimo/catalog/minio.properties:/opt/presto-server/etc/catalog/minio.properties
```
minio as a catalog?

wanglinsong commented on running locally/docker-prestissimo/docker-compose.yml:
```diff
@@ -0,0 +1,119 @@
+      - ./prestissimo/coordinator/jvm.config:/opt/presto-server/etc/jvm.config
+      - ./prestissimo/coordinator/node.properties:/opt/presto-server/etc/node.properties
+      - ./prestissimo/catalog/jmx.properties:/opt/presto-server/etc/catalog/jmx.properties
+      - ./prestissimo/catalog/tpcds.properties:/opt/presto-server/etc/catalog/tpcds.properties
```
tpcds is not supported now on prestissimo

wanglinsong commented on running locally/docker-prestissimo/docker-compose.yml:
```diff
@@ -0,0 +1,119 @@
+      - minio
+      - hadoop
+    environment:
+      - AWS_ACCESS_KEY=FEEF05540F5D8BE7EB62
```
is this the release value?

minhancao replied:
I could change this to be like this https://github.com/prestodb/prestorials/blob/main/docker-compose/docker-compose.yaml:
```
environment:
      - AWS_ACCESS_KEY=${AWS_ACCESS_KEY}
      - AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
```

What do you think?

wanglinsong commented on running locally/docker-prestissimo/docker-compose.yml:
```diff
@@ -0,0 +1,119 @@
+      - ./prestissimo/workers/1/config_velox.properties:/opt/presto-server/etc/config.properties
+      - ./prestissimo/workers/jvm.config:/opt/presto-server/etc/jvm.config
+      - ./prestissimo/workers/1/node_velox.properties:/opt/presto-server/etc/node.properties
+      # - ./prestissimo/catalog/jmx.properties:/opt/presto-server/etc/catalog/jmx.properties
```
please just delete them

minhancao replied:
Removed

wanglinsong commented on running locally/docker-prestissimo/docker-compose.yml:
```diff
@@ -0,0 +1,119 @@
+    depends_on:
+      - minio
+      - hadoop
+    entrypoint: |
```
entrypoint override?

minhancao replied:
This was from Christian's presto-minio repo: https://github.com/czentgr/presto-minio/blob/main/docker-compose.yml

This mc-job container was just meant to be ran to create 2 buckets and load some data inside them in MinIO storage and then be able to query it from presto-cli. It was meant just for fun so I'll remove this container.

wanglinsong commented on running locally/docker-prestissimo/prestissimo/catalog/jmx.properties:
```diff
@@ -0,0 +1,5 @@
+
+jmx.dump-period=10s
+jmx.dump-tables=java.lang:type=Runtime,com.facebook.presto.execution.scheduler:name=NodeScheduler
+jmx.max-entries=86400
```
please always have a newline at the bottom of a file

minhancao replied:
Done

wanglinsong commented on running locally/docker-prestissimo/prestissimo/catalog/minio.properties:
```diff
@@ -0,0 +1,11 @@
+connector.name=hive-hadoop2
```
please rename this as hive.properties

minhancao replied:
Done

wanglinsong reviewed (changes requested):
Please do run this on both macOS and Linux, to make sure it works.

minhancao commented:
@wanglinsong I have addressed your PR comments, please review and let me know if any changes are needed, thanks!

I have ran this docker setup on my Mac with Intel cpu and it works. Have not tried it on Linux yet will let you know.

wanglinsong reviewed (changes requested):
can you also rename folder 'running locally/docker-prestissimo' to just 'docker-compose-native'? 
'running locally' with space seems strange.

wanglinsong commented on running locally/docker-prestissimo/prestissimo/coordinator/log.properties:
```diff
@@ -0,0 +1 @@
+com.facebook.presto=DEBUG
```
please make sure there is a newline at the end of all files.

minhancao replied:
Done

wanglinsong commented on running locally/docker-prestissimo/prestissimo/workers/1/config_velox.properties:
```diff
@@ -0,0 +1,4 @@
+coordinator=false
+http-server.http.port=8081
+discovery.uri=http://172.19.0.4:8080
+presto.version=0.282-fadfd95
```
please rename this file to just config.properties

Use the version 0.282.
Any reason to use 8081? The yaml file does not expose it to the host.

minhancao replied:
Changed presto.version to be 0.282 for both the worker's config.properties and coordinator's config.properties since both have to match for it to work.

Also changed 8081 to 8080.

wanglinsong commented on running locally/docker-prestissimo/readme.md:
```diff
@@ -0,0 +1,13 @@
+
+It does take some time for the Prestissimo worker to be hooked up with the Presto coordinator. You can check if it's connected or not by going to http://localhost:8080/ui/ and it should show a dashboard with current active workers and other stats.
+
+To bring the Docker containers down, run `docker-compose down`.
```
docker compose up/down is the new command format.

minhancao replied:
Changed to new command format

wanglinsong reviewed (changes requested):
please also test the tpch and hive catalogs.

wanglinsong commented:
Also, please squash all updates into one commit. We will do rebase and merge.

minhancao commented:
@wanglinsong I have squashed all commits into one commit. I have also tested querying the tpch and hive catalogs on my Mac and it works.

wanglinsong reviewed (approved):
looks good to me
